### ISSUE TITLE
Problem installing

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#400]
Migrated from [rt.perl.org#1235](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=1235) (status was 'resolved')

Searchable as RT1235$

**Comment by p5pRT** [comment]
### From dminogue@rand.com

Thanks\,

Dave

\# make
make&#8203;: Warning&#8203;: Both \`makefile' and \`Makefile' exist
\`sh  cflags libperl\.a miniperlmain\.o\`  miniperlmain\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a perl\.o\`  perl\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a gv\.o\`  gv\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a toke\.o\`  toke\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a perly\.o\`  perly\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a op\.o\`  op\.c
&nbsp;         CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a regcomp\.o\`  regcomp\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a dump\.o\`  dump\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a util\.o\`  util\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a mg\.o\`  mg\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a byterun\.o\`  byterun\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a hv\.o\`  hv\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a av\.o\`  av\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a run\.o\`  run\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a pp\_hot\.o\`  pp\_hot\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a sv\.o\`  sv\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a pp\.o\`  pp\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a scope\.o\`  scope\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a pp\_ctl\.o\`  pp\_ctl\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a pp\_sys\.o\`  pp\_sys\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a doop\.o\`  doop\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a doio\.o\`  doio\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a regexec\.o\`  regexec\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a taint\.o\`  taint\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a deb\.o\`  deb\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a universal\.o\`  universal\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a globals\.o\`  globals\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
\`sh  cflags libperl\.a perlio\.o\`  perlio\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
rm \-f libperl\.a
/usr/ccs/bin/ar rcu libperl\.a perl\.o  gv\.o toke\.o perly\.o op\.o regcomp\.o
dump\.o util\.o mg\.o byterun\.o hv\.o av\.o run\.o pp\_hot\.o sv\.o pp\.o scope\.o
pp\_ctl\.o pp\_sys\.o doop\.o doio\.o regexec\.o taint\.o deb\.o universal\.o
globals\.o perlio\.o 
gcc  \-L/usr/local/lib \-o miniperl miniperlmain\.o libperl\.a \-lgdbm \-ldb
Undefined                       first referenced
 symbol                             in file
socket                              libperl\.a\(pp\_sys\.o\)
getpeername                         libperl\.a\(pp\_sys\.o\)
accept                              libperl\.a\(pp\_sys\.o\)
log                                 libperl\.a\(pp\.o\)
pow                                 libperl\.a\(pp\.o\)
socketpair                          libperl\.a\(pp\_sys\.o\)
send                                libperl\.a\(pp\_sys\.o\)
bind                                libperl\.a\(pp\_sys\.o\)
setsockopt                          libperl\.a\(pp\_sys\.o\)
sqrt                                libperl\.a\(pp\.o\)
floor                               libperl\.a\(pp\.o\)
getsockopt                          libperl\.a\(pp\_sys\.o\)
atan2                               libperl\.a\(pp\.o\)
sendto                              libperl\.a\(pp\_sys\.o\)
exp                                 libperl\.a\(pp\.o\)
shutdown                            libperl\.a\(pp\_sys\.o\)
getsockname                         libperl\.a\(pp\_sys\.o\)
recvfrom                            libperl\.a\(pp\_sys\.o\)
sin                                 libperl\.a\(pp\.o\)
listen                              libperl\.a\(pp\_sys\.o\)
cos                                 libperl\.a\(pp\.o\)
connect                             libperl\.a\(pp\_sys\.o\)
ld&#8203;: fatal&#8203;: Symbol referencing errors\. No output written to miniperl
\*\*\* Error code 1
make&#8203;: Fatal error&#8203;: Command failed for target \`miniperl'

myconfig


---
Summary of my perl5 \(5\.0 patchlevel 5 subversion 3\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=solaris\, osvers=2\.5\.1\, archname=sun4\-solaris
&nbsp;    uname='sunos roger 5\.5\.1 generic\_103640\-27 sun4u sparc sunw\,ultra\-1 '
&nbsp;    hint=previous\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=undef useperlio=undef d\_sfio=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc'\, optimize='\-g'\, gccversion=2\.8\.1
&nbsp;    cppflags='\-I/usr/local/include'
&nbsp;    ccflags ='\-I/usr/local/include'
&nbsp;    stdchar='unsigned char'\, d\_stdstdio=define\, usevfork=false
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    alignbytes=8\, usemymalloc=n\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='ld'\, ldflags =' \-L/usr/local/lib '
&nbsp;    libpth=/usr/local/lib /usr/lib
&nbsp;    libs=\-lgdbm \-ldb
&nbsp;    libc=/usr/lib/libc\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_none\.xs\, dlext=none\, d\_dlsymun=undef\, ccdlflags=''
&nbsp;    cccdlflags=''\, lddlflags=''

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

On Thu\, Aug 19\, 1999 at 04&#8203;:20&#8203;:11PM \-0400\, Minogue\, Dave wrote&#8203;:
> Here is the output after of make\, and below that is the output from the
> myconfig shell script\.
> I've spent nearly 2 days trying to install this without any luck\.

Dave\,

&nbsp;    You seem to have overridden a few defaults for the worse\.  The
simple remedy is&#8203;:

&nbsp;    \# make distclean
&nbsp;    \# Configure \-des \-Dcc=gcc
&nbsp;    \# make

&nbsp;    If this doesn't work\, then I've underestimated your problem\.  ;^\)

&nbsp;    Once you have this working\, then you may wish to run \`Configure'
interactively and tweak some of the options a bit more gingerly\.

> \# make
> make&#8203;: Warning&#8203;: Both \`makefile' and \`Makefile' exist
> \`sh  cflags libperl\.a miniperlmain\.o\`  miniperlmain\.c
>           CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   

&nbsp;    A default CCCMD will look more like&#8203;:

&nbsp;        gcc \-B/usr/ccs/bin/ \-DPERL\_CORE \-c \-I/usr/local/include \-O

&nbsp;    Your version might work\, but it suggests shenanigans to me\.

>   \[ \. \. \. \]
> 
> \`sh  cflags libperl\.a perlio\.o\`  perlio\.c
>           CCCMD =  gcc \-DPERL\_CORE \-c \-I/usr/local/include \-g   
> rm \-f libperl\.a
> /usr/ccs/bin/ar rcu libperl\.a perl\.o  gv\.o toke\.o perly\.o op\.o regcomp\.o
> dump\.o util\.o mg\.o byterun\.o hv\.o av\.o run\.o pp\_hot\.o sv\.o pp\.o scope\.o
> pp\_ctl\.o pp\_sys\.o doop\.o doio\.o regexec\.o taint\.o deb\.o universal\.o
> globals\.o perlio\.o 
> gcc  \-L/usr/local/lib \-o miniperl miniperlmain\.o libperl\.a \-lgdbm \-ldb

&nbsp;    There are some very important libraries missing from here\.  This
should look more like&#8203;:

&nbsp;    gcc \-B/usr/ccs/bin/  \-L/usr/local/lib \-o miniperl miniperlmain\.o libperl\.a
&nbsp;        \-lsocket \-lnsl \-lgdbm \-ldb \-ldl \-lm \-lc \-lcrypt 
&nbsp;    
> Undefined                       first referenced
>  symbol                             in file
> socket                              libperl\.a\(pp\_sys\.o\)

&nbsp;    \[ \. \. \. \]

> connect                             libperl\.a\(pp\_sys\.o\)
> ld&#8203;: fatal&#8203;: Symbol referencing errors\. No output written to miniperl
> \*\*\* Error code 1

&nbsp;     \. \. \. and these are all symbols from \-lm\, \-lsocket\, etc\.

> make&#8203;: Fatal error&#8203;: Command failed for target \`miniperl'
> 
> myconfig
> \-\-\-\-\-\-\-\-\-\-\-\-\-
> Summary of my perl5 \(5\.0 patchlevel 5 subversion 3\) configuration&#8203;:
>   Platform&#8203;:
>     osname=solaris\, osvers=2\.5\.1\, archname=sun4\-solaris

&nbsp;    I've compared this to a Solaris 2\.6 installation\, but most of the
information will be the same\.

&nbsp;    Best of luck\,
\* Kurt Starsinic \(Kurt\.Starsinic@&#8203;isinet\.com\) \-\-\-\-\-\-\-\-\- Technical Specialist \*
| \`The suddenness of the leap from hardware to software cannot but produce  |
|  a period of anarchy and collapse in existing establishments \. \. \. \.'     |
|                            \-\- Marshall McLuhan                            |
Institute for Scientific Information                   http&#8203;://www\.isinet\.com/

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

>>>>> "Minogue\," == Minogue\, Dave \<dminogue@&#8203;rand\.com> writes&#8203;:
&nbsp;    Minogue\,> Here is the output after of make\, and below that is the
&nbsp;    Minogue\,> output from the myconfig shell script\.  I've spent
&nbsp;    Minogue\,> nearly 2 days trying to install this without any luck\.

For some reason\, you've removed the '\-lsocket \-lnsl' libraries from
the list of libraries used to link perl\. Don't Do That\.
 
Put simply\, on Solaris leave the Configure selected libraries alone\.

\-\- 
Stephen
 \-\-\- 
"If 8\-year\-old boys discharging loaded firearms into their own legs
isn't necessary to the maintenance of a well\-regulated militia\, I
don't know what is\." \- Randal Cummings as reported in The Onion\, 25/5/99
